src/service-worker.js

```diff
@@ -0,0 +1,62 @@
+/// <reference types="@sveltejs/kit" />
+import { build, files, version } from '$service-worker';
+
+// Create a unique cache name for this deployment
+const CACHE = `cache-${version}`;
+
+const ASSETS = [
+    ...build, // the app itself
+    ...files  // everything in `static`
+];
+
+self.addEventListener('install', (event) => {
+    // Create a new cache and add all files to it
+    async function addFilesToCache() {
+        const cache = await caches.open(CACHE);
+        await cache.addAll(ASSETS);
+    }
```

Comment:
Should we try catch this in case there are exceptions?

Reply:
I actually want to throw that error back up 👍🏻

Reply:
This is then handled by the browser